cmd/juju/application/deploy.go

```diff
@@ -387,80 +387,85 @@ type DeployCommand struct {
-unexpected behavior.
+ 1. the '--series' deploy command option
+ 2. the series stated in the charm URL
+ 3. for a bundle, the 'series' field in the charm spec
+ 4. for a bundle, the 'series' field at top level of bundle file
+ 5. the 'default-series' model key
+ 6. the top-most series specified in the charm's metadata file
+    (this sets the charm's 'preferred series' in the Charm Store)
+ 
+If the determined series is not supported by the charm an error is emitted. The
+'--force' option can be used to override this check:
 
-Local bundles are specified with a direct path to a bundle.yaml file.
-For example:
+  juju deploy charm --series xenial --force
 
-  juju deploy /path/to/bundle/openstack/bundle.yaml
+When charms that include LXD profiles are deployed the profiles are validated
+using a hardcoded allow/deny list. This check can be bypassed with the use of
+the '--force' option. However, doing so is not recommended as it can lead to
+unexpected behaviour.
+
+If an 'application name' is not provided, the application name used is the charm
+or bundle name. A user-supplied 'application name' must consist only of
+lower-case letters (a-z), numbers (0-9), and single hyphens (-). The name must
+begin with a letter and not have a group of all numbers follow a hyphen:
 
-If an 'application name' is not provided, the application name used is the
-'charm or bundle' name.  A user-supplied 'application name' must consist only of
-lower-case letters (a-z), numbers (0-9), and single hyphens (-).  The name must
-begin with a letter and not have a group of all numbers follow a hyphen.
-Examples:
   Valid:   myappname, custom-app, app2-scat-23skidoo
   Invalid: myAppName, custom--app, app2-scat-23, areacode-555-info
 
-Constraints can be specified by specifying the '--constraints' option. If the
-application is later scaled out with ` + "`juju add-unit`" + `, provisioned machines
-will use the same constraints (unless changed by ` + "`juju set-constraints`" + `).
+Use the '--constraints' option to specify hardware requirements for new machines.
+These become the application's default constraints (i.e. they are used if the
+application is later scaled out with the ` + "`add-unit`" + ` command). To overcome this
+behaviour use the ` + "`set-constraints`" + ` command to change the application's default
+constraints or add a machine (` + "`add-machine`" + `) with a certain constraint and then
+target that machine with ` + "`add-unit`" + ` by using the '--to' option.
+
+Use the '--device' option to specify GPU device requirements (with Kubernetes).
+The below format is used for this option's value, where the 'label' is named in
+the charm metadata file:
 
-Devices can be specified by specifying the '--device' option to deploy charms to a
-k8s cluster which require the use of a GPU (or many).
-Devices provided should be in format:
-	<label>=[<count>,]<device-class>|<vendor/type>[,<attributes>]
+  <label>=[<count>,]<device-class>|<vendor/type>[,<attributes>]
```

Comment:
I'm not seeing a `--device` option in `juju-2.4.4-xenial-amd64`. Should this (and prior para) be removed?

Reply:
This PR is for 2.5. For snaps, it is in the --edge channel.